Packs/Illumio/Integrations/IllumioCore/IllumioCore.py

```diff
@@ -0,0 +1,107 @@
+import demistomock as demisto
+import urllib3
+from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
+from CommonServerUserPython import *  # noqa
+from illumio import PolicyComputeEngine
+
+urllib3.disable_warnings()
+
+''' CONSTANTS '''
+
+MIN_PORT = 1
+MAX_PORT = 65535
+
+'''EXCEPTION CLASS'''
+
+
+class InvalidPortException(Exception):
+    """Exception class for Invalid port."""
+
+    def __init__(self, port):
+        self.message = "{} is an invalid value for port. Value must be in 1 to 65535.".format(port)
+        super().__init__(self.message)
+
+
+''' CLIENT CLASS '''
+
+
+class IllumioClient:
+    """IllumioClient class to interact with the service SDK."""
+
+    def __init__(self, base_url: str, port: int, api_user: str, api_key: str, proxy: dict, org_id: int) -> None:
+        """Construct IllumioClient class object.
+
+        Args:
+            base_url: the base url to establish connection with the illumio sdk.
+            port: the port number to establish connection.
+            api_user: the username to authenticate with the illumio sdk.
+            api_key: the api key to authenticate with the illumio sdk.
+            proxy: whether to use the demisto's proxy settings or not.
+            org_id: the id of the organization to be referenced.
+        """
+
+        self.pce = PolicyComputeEngine(url=base_url, port=port, org_id=org_id)
+        self.pce.set_proxies(http_proxy=proxy.get("http"), https_proxy=proxy.get('https'))
+        self.pce.set_credentials(api_user, api_key)
+
+
+def test_module(client: IllumioClient) -> str:
+    """Tests API connectivity and authentication.
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+
+    Args:
+        client: IllumioClient to be used.
+
+    Returns: 'ok' if test passed, anything else will fail the test.
+    """
+    response = client.pce.check_connection()
+    if response:
+        return 'ok'
+    raise ValueError("Failed to establish connection with provided credentials.")
+
+
+def main():
+    """Parse params and runs command functions."""
+    params = demisto.params()
+    api_user = params.get('api_user')
+    api_key = params.get('api_key')
+
+    port = arg_to_number(params.get('port'))
+    if port < MIN_PORT or port > MAX_PORT:  # type: ignore
+        raise InvalidPortException(port)
```

Comment:
It seems like we still have the same problem here - this will be an uncaught exception with no `return_error` call. Is there a reason we can't just wrap this whole function in the `try:` block?

Reply:
we though it will only raise the common exception error only, ignoring the other ones. But yes we'll enclose the whole function inside the `try` block